contracts/Market.sol

```diff
@@ -345,6 +365,95 @@ contract Market is
     }
   }
 
+  /**
+   * @notice Purchases removals on behalf of the Certificate contract in order to replace removals that have been
+   * released from an existing certificate.
+   * @dev Replacement removals are sent to the Certificate contract and an event is emitted by the Certificate contract
+   * to indicate the specific certificate on behalf of which removals are being replaced, as well as the IDs and amounts
+   * of the replacement removals.
+   *
+   * ##### Requirements:
+   *
+   * - Can only be used when this contract is not paused.
+   * - The caller must have the MARKET_ADMIN_ROLE.
+   * - The amount of removals to purchase must be less than or equal to the amount of removals available in the
+   * market.
+   *
+   * @param treasury The address of the treasury that will fund the replacement purchase.
+   * @param certificateId The ID of the certificate on behalf of which removals are being replaced.
+   * @param totalAmountToReplace The total amount of replacement removals to purchase.
+   * @param removalIdsBeingReplaced The removal ids that are being replaced.
+   * @param amountsBeingReplaced The amount of each removal id that is being replaced.
+   */
+  function replace(
+    address treasury,
+    uint256 certificateId,
+    uint256 totalAmountToReplace,
+    uint256[] memory removalIdsBeingReplaced,
+    uint256[] memory amountsBeingReplaced
+  ) external whenNotPaused onlyRole(MARKET_ADMIN_ROLE) {
+    if (_certificate.getPurchaseAmount(certificateId) == 0) {
+      revert CertificateNotYetMinted({tokenId: certificateId});
+    }
+    int256 _guaranteeDiscrepancy = _certificate.getGuaranteeDiscrepancy();
```

Comment:
This should be updated in accordance to the changes in your other PR.
https://github.com/nori-dot-eco/contracts/pull/497#pullrequestreview-1266121711